### PR TITLE
feat(client): ts-migrate .../Challenges/components/PrismFormatted.js

### PR DESCRIPTION
**Pull request by thecodingaviator**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

#42256

### CODE REVIEW

File: client/src/templates/Challenges/components/prism-formatted.tsx [hunk 1]
```diff
@@ -32,7 +32,4 @@ class PrismFormatted extends Component {
   }
 }
 
-PrismFormatted.displayName = 'PrismFormatted';
```
Comment (ShaunSHamilton) [inline client/src/templates/Challenges/components/prism-formatted.tsx:35]: Does getting rid of the `displayName` negatively affect the dev experience? I have never used it in personal projects.

Comment (thecodingaviator) [reply client/src/templates/Challenges/components/prism-formatted.tsx:35]: I’d say so. This is very helpful in debugging. I think I deleted it by mistake. Good catch!

File: client/src/templates/Challenges/components/prism-formatted.tsx [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
+interface PrismFormattedProps {
+  className?: string;
+  text: string;
+}
 
-class PrismFormatted extends Component {
-  componentDidMount() {
+class PrismFormatted extends Component<PrismFormattedProps> {
+  instructionsRef: React.RefObject<HTMLInputElement>;
+  componentDidMount(): void {
```
Comment (ShaunSHamilton) [inline client/src/templates/Challenges/components/prism-formatted.tsx:11]:
```suggestion
  displayName = 'PrismFormatted';
  instructionsRef: React.RefObject<HTMLInputElement>;
  componentDidMount(): void {
```

File: client/src/templates/Challenges/components/prism-formatted.tsx [hunk 3]
```diff
@@ -33,6 +33,5 @@ class PrismFormatted extends Component {
 }
 
 PrismFormatted.displayName = 'PrismFormatted';
```
Comment (ShaunSHamilton) [inline client/src/templates/Challenges/components/prism-formatted.tsx:35]:
```suggestion
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Linter does not like this method of setting class props. I am not too sure of the best semantic way to do this, because I might be tempted to do:
```typescript
readonly displayName = '...';
```

File: client/src/templates/Challenges/components/prism-formatted.tsx (hunk 2 above)
Comment (thecodingaviator) [reply client/src/templates/Challenges/components/prism-formatted.tsx:11]: From what I’ve seen a static declaration is usually used. I’ll make that change

**Review by raisedadead — COMMENTED** [review]
We have landed the `next` branch into `main`, and I have rebased and retargeted your pull request. Please make sure to sync with your local before you continue to make any further changes.

If you need assistance with that, let us know in the contributors chat or here.

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍

**Review by raisedadead — CHANGES_REQUESTED** [review]
Blocking this for all the renovate PRs to land. Merging this will unintentionally create a blocker for renovate. I will land this first thing in the morning.

**Review by raisedadead — APPROVED** [review]

**Pull request merged**